File: core/src/main/java/me/devsnox/planetsystem/core/commands/modules/HomeCommand.java
package me.devsnox.planetsystem.core.commands.modules;

import me.devsnox.planetsystem.api.player.PlanetPlayer;
import me.devsnox.planetsystem.core.commands.PlanetCommandModule;
import org.bukkit.GameMode;
import org.bukkit.entity.Player;

public class HomeCommand implements PlanetCommandModule {

    @Override
    public void execute(final PlanetPlayer player, final String[] args) {
        final Player target = player.getPlayer();
        target.setGameMode(GameMode.SURVIVAL);
        target.setFallDistance(0);
        target.teleport(player.getPlanet().getSpawnLocation().toBukkitLocation());
        player.getLogger().info("commands.home.teleport.success");
    }
}

File: core/src/main/kotlin/me/devsnox/planetsystem/core/commands/modules/HomeCommand.kt
package me.devsnox.planetsystem.core.commands.modules

import me.devsnox.planetsystem.api.location.toBukkitLocation
import me.devsnox.planetsystem.api.player.PlanetPlayer
import me.devsnox.planetsystem.core.commands.PlanetCommandModule
import org.bukkit.GameMode

class HomeCommand : PlanetCommandModule {

	override fun execute(player: PlanetPlayer, args: Array<String>) {
		val target = player.player
		target.gameMode = GameMode.SURVIVAL
		target.fallDistance = 0f
		target.teleport(player.planet.spawnLocation.toBukkitLocation())
		player.logger.info("commands.home.teleport.success")
	}
}
